sycl/source/detail/scheduler/scheduler2.cpp

```diff
@@ -21,7 +21,16 @@ namespace sycl {
 namespace detail {
 
 void Scheduler::waitForRecordToFinish(GraphBuilder::MemObjRecord *Record) {
-  for (Command *Cmd : Record->MLeafs) {
+  for (Command *Cmd : Record->MReadLeafs) {
+    Command *FailedCommand = GraphProcessor::enqueueCommand(Cmd);
+    if (FailedCommand) {
+      // TODO: What is the best way to handle failed command in Scheduler d'tor?
```

Comment:
I'm more on the throw runtime error version.

Reply:
I'm not sure about throwing exceptions in d'tors. Will we get error from static analyzers?